fdbclient/MasterProxyInterface.h

```diff
@@ -136,6 +137,10 @@ struct CommitTransactionRequest : TimedRequest {
 	void serialize(Ar& ar) { 
 		serializer(ar, transaction, reply, arena, flags, debugID);
 	}
+
+	void reportConflictingKeys(){
+		transaction.report_conflicting_keys = true;
```

Comment:
I don't think the way you're setting this transaction option is consistent with other transaction options. I think the following would be more consistent (and therefore easier to understand for existing users):
1. Remove this method from `CommitTransactionRequest` - it is not needed.
1. Add `reportConflictingKeys` to the `TransactionOptions` in `NativeAPI`  and make sure is set back to `false` when reset is called on it.
1. In `Transaction::commitMutations()` you can check whether this option was set and then set the option in the transaction object directly.

Reply:
Totally agree. My bad to forget this bad way to set it. Fixed it.